### PR TITLE
Don't show empty genrule filter labels when adding a new genrule

### LINKED ISSUE #808: Generation rule empty params showing
After creating a new generation rule, the empty parameter labels show up like:

Capture d’écran du 2023-03-14 16-08-29

Note that this only shows up when adding a new generation rule. once you save it and reload the page, the empty parameter labels arent shown (which is what is expected)

### PR DESCRIPTION
**Pull request by ryanlerch**
Resolves: #808

### CODE REVIEW

File: frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue [hunk 1]
```diff
@@ -41,6 +41,7 @@ const handleDeleteClicked = () => {
             v-if="
               f_params !== null &&
               f_params !== false &&
+              f_params !== undefined &&
```
Comment (abompard) [inline frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue]: Two things:
- testing for a variable being undefined in JS is usually done with `typeof variable == "undefined"`, I'm not sure why it is better than comparing to `undefined` directly like you do but maybe there are corner cases
- it feels like this whole test could be summed up as testing whether `f_params` is *truthy*, in JS there are several ways of converting a variable to a boolean, the one I've seen most is `!!variable` (two exclamation points). Would that work? I think we'd still need to test for empty array, but the three others should be covered.

Comment (ryanlerch) [reply frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue]: !!variable works perfectly. thanks

File: frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue [hunk 2]
```diff
@@ -32,16 +32,15 @@ const handleDeleteClicked = () => {
         :key="`${dest.protocol}:${dest.address}`"
       />
       <div>
+        {{ props.rule.filters }}
```
Comment (abompard) [inline frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue]: Looks like a forgotten debugging line

**Review by abompard — CHANGES_REQUESTED** [review]
